### PR TITLE
St endpoints definition

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

File: src/errors.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+exports.errnos = errnos;
+
+// Error messages.
+const errors = {
+  BAD_REQUEST     : 'Bad Request',
+  FORBIDDEN       : 'Forbidden',
+  INTERNAL_ERROR  : 'Internal Server Error',
+  UNAUTHORIZED    : 'Unauthorized'
+};
+exports.errors = errors;
```
Comment (ferjm) [inline src/errors.js]: Let's not anticipate any error code or message. For example,`ERRNO_INVALID_API_CLIENT_NAME` or `ERRNO_UNAUTHORIZED` will probably not be used here. Just clear this objects for now and we will be adding the errors as we need them.

File: src/errors.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// Model errors.
+const modelErrors = {
+  RECORD_ALREADY_EXISTS: 'RecordAlreadyExists',
+  UNSUPPORTED_AUTH_METHOD: 'UnsupportedAuthMethod'
+};
+exports.modelErrors = modelErrors;
```
Comment (ferjm) [inline src/errors.js]: Ditto.

File: src/routes/datastreams.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+let router = express.Router();
+
+/**
+ * Implementation of 8.3.4 "Datastreams"
+ * The OGC SensorThings API follows the ITU-T definition, i.e., with regard
+ * to the Internet of Things, a thing is an object of the physical world
+ * (physical things) or the information world (virtual things) that is capable
+ * of being identified and integrated into communication networks [ITU-T Y.2060].
+
+
```
Comment (ferjm) [inline src/routes/datastreams.js]: nit: extra line
nit: missing `*`

File: src/routes/datastreams.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ *      "Sensor@iot.navigationLink": "Datastreams(1)/Sensor",
+ *      "ObservedProperty@iot.navigationLink": "Datastreams(1)/ObservedProperty",
+ *      "Observations@iot.navigationLink": "Datastreams(1)/Observations",
+ *      "description": "This is a datastream measuring the temperature in an oven.",
+ *        "unitOfMeasurement": {
+ *        "name": "degree Celsius",
+ *        "symbol": "°C",
+ *        "definition": "http://unitsofmeasure.org/ucum.html#para-30"
+ *      },
+ *      "observationType": "http://www.opengis.net/def/observationType/OGC- OM/2.0/OM_Measurement",
```
Comment (ferjm) [inline src/routes/datastreams.js]: nit: lines shorter than 80 chars, please. Here and on the rest of the patch.

File: src/routes/featuresOfInterest.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+let router = express.Router();
+
+/**
+ * Implementation of 8.3.8 "FeaturesOfInterest"
+ * The OGC SensorThings API follows the ITU-T definition, i.e., with regard
+ * to the Internet of Things, a thing is an object of the physical world
+ * (physical things) or the information world (virtual things) that is capable
+ * of being identified and integrated into communication networks [ITU-T Y.2060].
+
+
```
Comment (ferjm) [inline src/routes/featuresOfInterest.js]: Same nit about extra line and missing `*`

File: src/routes/historicalLocations.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+let router = express.Router();
+
+/**
+ * Implementation of 8.3.3 "HistoricalLocation"
+ * The OGC SensorThings API follows the ITU-T definition, i.e., with regard
+ * to the Internet of Things, a thing is an object of the physical world
+ * (physical things) or the information world (virtual things) that is capable
+ * of being identified and integrated into communication networks [ITU-T Y.2060].
+
+
```
Comment (ferjm) [inline src/routes/historicalLocations.js]: Same nit about extra line and missing `*`

File: src/routes/locations.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+let router = express.Router();
+
+/**
+ * Implementation of 8.3.2 "Location"
+ * The OGC SensorThings API follows the ITU-T definition, i.e., with regard
+ * to the Internet of Things, a thing is an object of the physical world
+ * (physical things) or the information world (virtual things) that is capable
+ * of being identified and integrated into communication networks [ITU-T Y.2060].
+
+
```
Comment (ferjm) [inline src/routes/locations.js]: Ditto.

File: test/server.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+import sensorthings     from '../src/sensorthings';
+
+let app = express();
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(expressValidator());
+app.use(bodyParser.json());
+
+logger(app);
```
Comment (ferjm) [inline test/server.js]: I find the logger for tests too verbose. I'd prefer to remove it.

File: test/test_base.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import should     from 'should';
+import supertest  from 'supertest';
+
+import app        from './server';
+import sensorthings    from '../src/sensorthings';
```
Comment (ferjm) [inline test/test_base.js]: nit: align `from`

File: test/server.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+const endpointPrefix = '/api';
```
Comment (ferjm) [inline test/server.js]: You can probably remove this prefix.

File: test/test_base.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+import {
+  BAD_REQUEST,
+  errnos,
+  ERRNO_FORBIDDEN,
+  ERRNO_INVALID_API_CLIENT_NAME,
+  ERRNO_UNAUTHORIZED,
+  errors,
+  FORBIDDEN,
+  UNAUTHORIZED
+} from '../src/errors';
```
Comment (ferjm) [inline test/test_base.js]: These are not used.

File: test/test_base.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+const endpointPrefix = '/api';
+const server = supertest.agent(app);
+let token;
```
Comment (ferjm) [inline test/test_base.js]: `token` is not used.

File: test/test_things.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+  it('should exist', done => {
+    server.get(endpointPrefix + '/Things')
+      .expect('Content-type', /json/)
+      .expect(200)
+      .end((err, res) => {
+        res.status.should.be.equal(200);
+        done();
+      });
+  });
+});
```
Comment (ferjm) [inline test/test_things.js]: Same comment about unused imports from `../src/errors` and unused `token` in all test files.

File: package.json [hunk 14]
```diff
@@ -26,6 +29,24 @@
...
+    "coveralls": "^2.11.14",
+    "istanbul": "^0.4.5",
+    "mocha": "^3.1.0",
+    "nodemon": "^1.10.2",
+    "nyc": "^8.3.1",
+    "should": "^11.1.0",
+    "supertest": "^2.0.0",
+    "express-validator": "^2.20.10",
+    "body-parser": "^1.15.2",
+    "morgan-body": "^0.9.1"
```
Comment (ferjm) [inline package.json]: You don't need `morgan-body` any more :)

File: test/server.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import bodyParser       from 'body-parser';
+import express          from 'express';
+import expressValidator from 'express-validator';
+import logger           from 'morgan-body';
```
Comment (ferjm) [inline test/server.js]: Remove this one, please.

File: test/server.js [hunk 16]
```diff
@@ -0,0 +1,18 @@
+import bodyParser       from 'body-parser';
+import express          from 'express';
+import expressValidator from 'express-validator';
```
Comment (ferjm) [inline test/server.js]: We are not really using `expressValidator` so I guess we can get rid of it.

File: package.json [hunk 17]
```diff
@@ -26,6 +29,24 @@
...
-    "babel-preset-es2015": "^6.16.0"
+    "babel-preset-es2015": "^6.16.0",
+    "coveralls": "^2.11.14",
+    "istanbul": "^0.4.5",
+    "mocha": "^3.1.0",
+    "nodemon": "^1.10.2",
+    "nyc": "^8.3.1",
+    "should": "^11.1.0",
+    "supertest": "^2.0.0",
+    "express-validator": "^2.20.10",
```
Comment (ferjm) [inline package.json]: No need for `express-validator`

File: package.json [hunk 18]
```diff
@@ -26,6 +29,24 @@
...
+    "body-parser": "^1.15.2",
+    "morgan-body": "^0.9.1"
+  },
+  "config": {
+    "blanket": {
```
Comment (ferjm) [inline package.json]: Do we use blanket?

File: src/sensorthings.js [hunk 19]
```diff
@@ -1,8 +1,24 @@
...
+import observations    from './routes/observations';
+import observedProperties    from './routes/observedProperties';
 
 let router = express.Router();
 
 router.get('/', base);
+router.get('/Things', things);
+router.get('/Locations', locations);
+router.get('/HistoricalLocations', historicalLocations);
+router.get('/FeaturesOfInterest', featuresOfInterest);
```
Comment (ferjm) [inline src/sensorthings.js]: It's funny that all the endpoints are in plural except this one... It seems like a spec issue

Comment (albertopq) [inline src/sensorthings.js]: Well, it is actually in plural. Feature's'OfInterest :)

Comment (ferjm) [inline src/sensorthings.js]: Hah! You are right :D

**Pull request merged**